index.bs

```diff
@@ -181,96 +182,66 @@ partial interface Navigator {
 
-<div class="algorithm" data-algorithm="default-device-selection">
+<div class="algorithm" data-algorithm="xr-device-selection">
 
-Each time the [=list of XR devices=] changes the user agent should <dfn>select a default XR device</dfn> by running the following steps:
+Each time the [=list of XR devices=] changes the user agent should <dfn>select an XR device</dfn> by running the following steps:
 
-  1. Let |oldDefaultDevice| be the [=default device=].
-  1. If the [=list of XR devices=] is empty, set the [=default device=] to <code>null</code>.
-  1. If the [=list of XR devices=] contains one device set the [=default device=] to that device.
-  1. If the [=list of XR devices=] contains multiple devices set the [=default device=] to a device of the user agent's choosing.
-  1. If this is not the first time devices have been enumerated and |oldDefaultDevice| does not equal the [=default device=] queue a task that fires a simple event named {{devicechange}} on the {{XR}} object.
+  1. Let |oldDevice| be the current [=XR device=].
+  1. If the [=list of XR devices=] is empty, set the [=XR device=] to <code>null</code>.
+  1. If the [=list of XR devices=] contains one device set the [=XR device=] to that device.
+  1. If the [=list of XR devices=] contains multiple devices set the [=XR device=] to a device of the user agent's choosing.
+  1. If this is not the first time devices have been enumerated and |oldDevice| does not equal the [=XR device=] queue a task that fires a simple event named {{devicechange}} on the {{XR}} object.
 
 </div>
 
-NOTE: The user agent is allowed to use any criteria it wishes to [=select a default XR device=] when the [=list of XR devices=] contains multiple devices. For example, the user agent may always select the first item in the list, or provide settings UI that allows users to manage device priority. Ideally the algorithm used to select the default device is stable and will result in the same device being selected across multiple browsing sessions.
+NOTE: The user agent is allowed to use any criteria it wishes to [=select an XR device=] when the [=list of XR devices=] contains multiple devices. For example, the user agent may always select the first item in the list, or provide settings UI that allows users to manage device priority. Ideally the algorithm used to select the default device is stable and will result in the same device being selected across multiple browsing sessions.
 
-<div class="algorithm" data-algorithm="request-device">
+<div class="algorithm" data-algorithm="ensure-device-selected">
 
-The page can request a device by calling the <dfn method for="XR">requestDevice()</dfn> method on the {{Navigator/xr}} object. When invoked it MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
+Any time an [=XR device=] is needed by an algorithm it can <dfn>ensure an XR device is selected</dfn> by running the following steps:
```

Comment:
This seems to imply that it might change from call to call even after a session has been established..

Reply:
I followed it, though yes... a bit convoluted :)
As a sanity check, are we certain about the ordering of those last three steps?  As in, can we think of any reason why we might want to fire the devicechange before shutting down the session?

Reply:
I gave some thought to that, and the primary reason I came up with for the current order is that some developers may try and create a session in response to the event. We wouldn't want to end it immediately after, so this order simply makes the logic cleaner.